### PR TITLE
Add Glossary section to User Guide

### LINKED ISSUE #165: Add Glossary section to User Guide

### PR DESCRIPTION
**Pull request by Ferusel**
Closes #165

Adds the Glossary section to the User Guide.

### CODE REVIEW

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
Please reconfigure your linter to not add hard line breaks in Markdown files. It violates the coding standard.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -781,4 +794,75 @@ Something goes here...
 
 ## Glossary
```
Comment (Eugenetanwl3881) [inline docs/UserGuide.md:795]: Perhaps, including some of the technical terms present will be useful:

- Command line
- syntax
- parameters/arguments 

These are non-exhaustive. Add more if you think I have missed out any. If you think they have been specified clearly enough in the UG, it is fine to leave them out.

**Comment by Ferusel** [comment]
I reverted the changes made from intelliJ's linter and used the markdown linter specified in #166. I also git push force'd following the [instructions here](https://github.com/AY2223S1-CS2103T-W16-2/tp/pull/166) FYI.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**